02_tidy-evaluation-metaprogramming.Rmd

```diff
@@ -1,13 +1,261 @@
 # Tidy Evaluation & Metaprogramming
 
+```{r, echo=FALSE}
+knitr::opts_chunk$set(
+  eval = FALSE,
+  echo = TRUE
+)
+```
+
+
 **Learning objectives:**
 
-- THESE ARE NICE TO HAVE BUT NOT ABSOLUTELY NECESSARY
+- Why data masking exists, and how it works, in theory
+- Patterns for using data masking in practice
+
+## Problem {-}
+
+### Part 1: reference {-}
+
+- Data frame columns are not objects in the global environment...
+- ... but we want to address them as such
+
+```{r}
+# proper search path: from environment object to column
+mean(mtcars$cyl + mtcars$am)
+
+# improper search: this is not the object you seek
+mean(mtcars$cyl + mtcars$am)
```

Comment:
was this meant to be `mean(cyl + am)` ?

Reply:
You're absolutely right. This was a mistake I noticed during the presentation, and that I failed to correct subsequently.

Reply:
```suggestion
mean(cyl + am)
```